mod/01_rating/set/type/metric_value.rb

```diff
@@ -50,18 +50,23 @@ def create_source
   Env.params[:sourcebox] = 'true'
   value = remove_subfield 'value'
   if (sub_source_card = subfield('source'))
-    new_source_card = sub_source_card.subcard('new_source')
-    source_subcards = {}
-    new_source_card.subcards.each_with_key do |subcard, key|
-      subcard_key = subcard.tag.key
-      if key == 'file'
-        source_subcards["+#{subcard_key}"] = { file: subcard.file.file, type_id: subcard.type_id }
-      else
-        source_subcards["+#{subcard_key}"] = { content: subcard.content, type_id: subcard.type_id }          
+    source_card = if (new_source_card = sub_source_card.subcard('new_source'))
+      source_subcards = {}
+      new_source_card.subcards.each_with_key do |subcard, key|
+        subcard_key = subcard.tag.key
+        if key == 'file'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.